src/editor/StatusSidebar.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * Options that appear in the "Status & Visibility" sidebar panel.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { PluginPostStatusInfo } from '@wordpress/edit-post';
+import { useEffect } from '@wordpress/element';
+import { ToggleControl } from '@wordpress/components';
+
+const StatusSidebar = ( {
+	createNotice,
+	frequency,
+	newspack_popups_is_sitewide_default,
+	onMetaFieldChange,
+	placement,
+	removeNotice,
+	onSitewideDefaultChange,
+} ) => {
+	const isTest = 'test' === frequency;
+	const isInline = 'inline' === placement;
+
+	const createTestNotice = () => {
+		createNotice(
+			'warning',
+			__(
+				'Test Mode Enabled: In "Test Mode" logged-in admins will see the campaign every time, and non-admins will never see them.',
+				'newspack-popups'
+			),
+			{
+				id: 'newspack-popups__test-mode',
+				isDismissible: true,
+				type: 'default',
+			}
+		);
+	};
+
+	useEffect(() => {
+		if ( isTest ) {
+			createTestNotice();
+		}
+	}, [ isTest ]);
+
+	return (
+		<PluginPostStatusInfo>
+			<div className="newspack-popups__status-options">
+				<ToggleControl
+					checked={ isTest }
+					label={ __( 'Test Mode', 'newspack-popups' ) }
+					help={ __(
+						'In "Test Mode" logged-in admins will see the campaign every time, and non-admins will never see them.',
+						'newspack-popups'
+					) }
+					onChange={ value => {
+						if ( value ) {
+							createTestNotice();
```

Comment:
Nitpick: That's already handled in the `useEffect` hook, where the removal of the notice should also take place.

Reply:
Good call—updated in 8fdc7d7.